### PR TITLE
[DOCS] Reformats count API

### PR DESCRIPTION
**Pull request by szabosteve**
Relates to elastic/docs#937 and https://github.com/elastic/elasticsearch/issues/45621.

This PR updates the count API to align with the new [API reference template](https://github.com/elastic/docs/blob/master/shared/api-ref-ex.asciidoc).

Resources:
* [Count API spec](https://github.com/elastic/elasticsearch/blob/master/rest-api-spec/src/main/resources/rest-api-spec/api/count.json)
* [Count API doc](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-count.html)

### CODE REVIEW

File: docs/reference/search/count.asciidoc [hunk 1]
```diff
@@ -1,11 +1,91 @@
...
 === Count API
 
-The count API allows to easily execute a query and get the number of
-matches for that query. It can be executed across one or more indices.
-The query can either be provided using a simple query string as a
-parameter, or using the <<query-dsl,Query DSL>> defined within the request
-body. Here is an example:
+Gets the number of matches for a search query.
+
+[source,js]
```
Comment (jrodewig) [inline docs/reference/search/count.asciidoc]:
```suggestion
[source,console]
```

This recently changed with #46189.

File: docs/reference/search/count.asciidoc [hunk 2]
```diff
@@ -1,11 +1,91 @@
...
+--------------------------------------------------
+GET /twitter/_count?q=user:kimchy
+--------------------------------------------------
+//CONSOLE
```
Comment (jrodewig) [inline docs/reference/search/count.asciidoc]:
```suggestion
// TEST[setup:twitter]
```

This is likely causing the CI to fail.

`//CONSOLE`  is no longer needed with `[source,console].

File: docs/reference/search/count.asciidoc [hunk 3]
```diff
@@ -1,11 +1,91 @@
...
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=analyze_wildcard]
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=default_operator]
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=df]
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=expand-wildcards]
+
+`ignore_throttled`::
```
Comment (jrodewig) [inline docs/reference/search/count.asciidoc]: Can we move this to `common-parms.asciidoc`?

Looks like this parameter is also used here:

- https://www.elastic.co/guide/en/elasticsearch/reference/master/search-template.html#search-template-api-query-params
- https://www.elastic.co/guide/en/elasticsearch/reference/master/search-search.html#search-search-api-query-params

File: docs/reference/search/count.asciidoc [hunk 4]
```diff
@@ -1,11 +1,91 @@
...
+  (Optional, boolean) If `true`, concrete, expanded or aliased indices are 
+  ignored when throttled.
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=index-ignore-unavailable]
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=lenient]
+
+`min_score`::
+  Sets the minimum `_score` value that documents must have to be included in the
```
Comment (jrodewig) [inline docs/reference/search/count.asciidoc]:
```suggestion
(Optional, float)
  Sets the minimum `_score` value that documents must have to be included in the 
```

File: docs/reference/search/count.asciidoc [hunk 5]
```diff
@@ -1,11 +1,91 @@
...
+
+`query`::
+  (Optional, <<query-dsl,query object>>) Defines the search definition using the 
+  <<query-dsl,Query DSL>>.
+
+
+[[search-count-api-example]]
+==== {api-examples-title}
 
 [source,js]
```
Comment (jrodewig) [inline docs/reference/search/count.asciidoc]:
```suggestion
[source,console]
```

This is actually required as the `//CONSOLE` comment was deleted below this.

File: docs/reference/search/count.asciidoc [hunk 6]
```diff
@@ -23,13 +103,9 @@ GET /twitter/_count
...
-
-NOTE: The query being sent in the body must be nested in a `query` key, same as
-the <<search-search,search api>> works
 
-Both examples above do the same thing, which is count the number of
-tweets from the `twitter` index for a certain user. The result is:
+Both examples above do the same: count the number of tweets from the `twitter` 
+index for a certain user. The API returns the following response:
 
 [source,js]
```
Comment (jrodewig) [inline docs/reference/search/count.asciidoc]:
```suggestion
[source,console-result]
```

You can also remove the `// TESTRESPONSE` comment below this.

**Review by jrodewig — CHANGES_REQUESTED** [review]
Left a few comments that should help fix the CI.

File: docs/reference/search/count.asciidoc [hunk 7]
```diff
@@ -1,11 +1,88 @@
...
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=analyzer]
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=analyze_wildcard]
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=default_operator]
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=df]
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=expand-wildcards]
```
Comment (jrodewig) [inline docs/reference/search/count.asciidoc]:
```suggestion
include::{docdir}/rest-api/common-parms.asciidoc[tag=expand-wildcards]
+
Defaults to `open`.
```

**Review by jrodewig — APPROVED** [review]
Left a few minor comments but generally LGTM.

**Comment by szabosteve** [comment]
Backported to 7.x, 7.4, and 7.3.

**Pull request merged**